arch(claim-task-id): blocked-by:tNNN auto-emission collides with pre-flight label validation; first-use UX wall in fresh consumer repos

## Description

`claim-task-id.sh` aborts (exit 3) on consumer repos that don't pre-seed per-task `blocked-by:t<NNN>` labels, when the user's `--description` references a predecessor task. The auto-emission introduced in t2823 / [PR #20839](https://github.com/marcusquinn/aidevops/pull/20839) (detect "blocked-by: t<NNN>" in description text → apply matching label) collides with the pre-flight label-validation guard introduced in t2800 / [PR #20744](https://github.com/marcusquinn/aidevops/pull/20744) (refuse to advance counter if any requested label doesn't exist). The collision creates a hard wall: the helper detects the predecessor reference, decides to apply the label, and aborts because the label doesn't exist — without an automatic recovery path.

This is filed as an architecture discussion, not a code-fix prescription, because the right resolution depends on framework design intent that maintainers are best placed to weigh.

## Expected Behavior

The user's intent is "this task is blocked by t<NNN>". The helper should record that intent without forcing the user to manually create a label, run with `--no-blocked-by` (which loses the label entirely), or hand-create a label they may not have known they needed.

What the right behaviour IS depends on a deliberate choice between three reasonable options (see Considerations below).

## Reproducer

**Symptom command** (real session, 2026-04-28 in `robstiles/qs-agency`):

```bash
~/.aidevops/agents/scripts/claim-task-id.sh \
  --repo-path ~/Git/qs-agency \
  --title "align 6 downstream briefs with locked t324 enum counts (16/11/11)" \
  --description "$(cat description-with-blocked-by-t324.md)" \
  --labels "deep-research,tier:standard,auto-dispatch,blocked-by:t324"
```

The description contained the line `**blocked-by:** t324 PR #2106 must merge first` in its Dependencies section.

**Actual output**:

```
[INFO] Loading project config from .aidevops.json
[WARN] GH#20834: Detected predecessor reference(s) in description; auto-emitting blocked-by:t324. Override with --no-blocked-by.
[INFO] Using remote: origin, counter branch: main
[INFO] Detected platform: github
[ERROR] Pre-flight label validation failed — invalid label(s): 'blocked-by:t324'
[ERROR]   Run: gh label list --repo "robstiles/qs-agency" | grep -i '<name>'
[ERROR]   Claim aborted — counter NOT advanced.
```

(Exit code 3.)

The error message tells the user how to *check* the label, not how to *fix* the situation. The helper has detected a real predecessor reference, intends to record it, and aborts — leaving the user to decide between three non-obvious workarounds, none of which are mentioned in the error output.

## Steps to Reproduce

1. In any consumer repo that does not have `blocked-by:t<NNN>` labels (i.e., any repo that hasn't been hand-seeded with these labels — most consumer repos at first use).
2. Run `claim-task-id.sh` with `--description` containing any predecessor reference text the t2823 detector recognises (`blocked-by: t<NNN>`, `Follow-up from t<NNN>`, `tracked in #<NNN>`, `after t<NNN> ships|merges|lands`).
3. Helper aborts with `Pre-flight label validation failed`. Counter is not advanced (correct), but no progress is made and the user has no clear next step.

## Workarounds Applied

In the failing session I tried three:

1. `--no-blocked-by` — works, but discards the predecessor information from labels entirely. Body text still contains the blocker reference, and `pulse-dep-graph.sh` (line 541-543) reads body text directly, so the dependency is still tracked for dispatch ordering. But the `blocked-by:t<NNN>` *label* — used by `pulse-issue-reconcile-actions.sh` and other consumers as a fast-filter index — is gone. This is what I shipped on the failing session.

2. `--labels "...,status:blocked"` (substituted a generic blocking label that DOES exist in qs-agency) — works, blocks dispatch, but loses the per-blocker specificity. `status:blocked` is one bit; `blocked-by:t324` carries the blocker identity.

3. `gh label create blocked-by:t324 --repo robstiles/qs-agency` then re-run — works, but requires the user to know the label is needed before they hit the wall, and creates per-task label proliferation that nobody decided to opt into.

The third option is the one a maintainer-aware user would converge on, but the framework gives them no guidance toward it from the error message.

## Context

The relevant interacting commits, in order:

| Commit | Task | Effect |
|---|---|---|
| [#20247](https://github.com/marcusquinn/aidevops/pull/20247) | t2548 (GH#20224) | Split `claim-task-id.sh` into counter and issue sub-libraries; introduced `_ensure_todo_entry_written` |
| [#20744](https://github.com/marcusquinn/aidevops/pull/20744) | t2800 | Added pre-flight label validation; abort with exit 3 before counter advance if any requested label doesn't exist |
| [#20839](https://github.com/marcusquinn/aidevops/pull/20839) | t2823 (GH#20834) | Auto-emit `blocked-by:t<NNN>` labels when predecessor references appear in `--description` |
| [#20863](https://github.com/marcusquinn/aidevops/pull/20863) | (follow-up) | Case-insensitive normalisation in `_detect_predecessor_refs` |

Each commit is individually correct. The collision emerges from their composition: t2800 says "every requested label must exist or we refuse"; t2823 says "any predecessor reference in description text becomes a requested label". Whether a given consumer repo has the corresponding `blocked-by:t<NNN>` label is path-dependent on whether someone has filed prior cross-task work in that repo, which is unrelated to whether the user's current intent is valid.

`blocked-by:t<NNN>` labels are NOT auto-created by `aidevops init` or any other framework setup step. They accumulate organically when users hit this wall and create them by hand. The first user of cross-task references in a fresh consumer repo will always hit this.

The framework already reads blocker info from body text (`pulse-dep-graph.sh:541-543` for `blocked-by:t<NNN>` and `blocked-by:#<NNN>`; `pulse-issue-reconcile-actions.sh:149` for `blocked-by:` patterns). Labels are a fast-filter index, not the source of truth. So the cost of "no label, body text only" is real (slower filtering, label-based pulse logic loses the specificity) but not catastrophic (dependencies still resolve).

## Considerations for the Fix Path

Three options, each with trade-offs the maintainer is best placed to weigh:

### Option A — Auto-create missing `blocked-by:t<NNN>` labels in `claim-task-id.sh`

When pre-flight detects a missing `blocked-by:t<NNN>` label requested via t2823 auto-emission, create the label automatically (`gh label create` with a sensible default colour and description) and proceed.

- **Pro**: zero user-visible friction. The label is a deterministic function of the predecessor task ID, so auto-creation is safe.
- **Pro**: aligns with how `tier:*`, `origin:*`, and `status:*` labels are handled at framework init — the label catalogue grows with use, no manual seeding required.
- **Con**: per-task label proliferation. A repo that has worked on 200 tasks accumulates 200 `blocked-by:t<NNN>` labels, most of which become irrelevant once the predecessor merges. Pruning needs a separate sweep.
- **Con**: changes the trust model of pre-flight validation. Currently t2800 means "labels are vetted"; auto-create weakens that.
- **Implementation surface**: small. `claim-task-id-issue.sh` `_validate_labels_pre_flight` (or wherever t2800's check lives) gains an "auto-create exception class" for `blocked-by:t<NNN>` patterns.

### Option B — Skip label application when label doesn't exist; warn but continue

When a `blocked-by:t<NNN>` label is requested via auto-emission and doesn't exist, log a warning ("blocker recorded in body text but no `blocked-by:t<NNN>` label exists in this repo; create with `gh label create blocked-by:t<NNN>` to enable label-based filtering"), drop the label from the apply set, and continue with the claim.

- **Pro**: respects user intent (claim proceeds), preserves t2800's "labels are vetted" trust model (we don't invent labels), and the body text still encodes the dependency for the consumers that read it.
- **Pro**: the warning trains the user toward Option A by hand if they want labelled blockers.
- **Con**: silently degrades the `blocked-by:t<NNN>` label semantic for consumers that DO read the label fast path. If the user expected dispatch-blocking via label, they don't get it.
- **Con**: makes auto-emission a "best effort" feature, which is a different contract from how it's documented.

### Option C — Document the requirement; provide a setup helper

Add to the framework docs that consumer repos should seed `blocked-by:t*` labels on initialization, and provide `aidevops init --seed-blocked-by-labels` (or similar) that creates the catalogue. Keep the current pre-flight refusal.

- **Pro**: simplest framework surface — no new logic, just docs and a setup step.
- **Pro**: keeps the trust model intact.
- **Con**: still a per-task-id label catalogue (200 task IDs → 200 labels) — same proliferation as Option A, just with a different trigger.
- **Con**: shifts the cost to the user. New consumer repos still hit the wall the first time, just with better docs.
- **Con**: doesn't help repos that have already been initialised (most of them).

### Sub-option for any of A/B/C — better error message

Whatever path is chosen, the current error message is unhelpful. It should at minimum say:

```
Pre-flight label validation failed — label 'blocked-by:t324' does not exist in this repo.

This label was auto-emitted by t2823 from a predecessor reference in your --description.
Options:
  (a) Create the label and re-run: gh label create blocked-by:t324 --repo <slug>
  (b) Skip the label: re-run with --no-blocked-by (dependency still recorded in body text)
  (c) Use a different blocking label that exists, e.g. status:blocked
```

This is independently shippable regardless of which Option A/B/C path is chosen.

## My Lean (Optional)

Option B — skip-with-warning — is what I'd choose if it were my call:

- It respects the t2800 trust model (we don't invent labels) while not stalling productive work.
- The warning is the right place to teach the user about Option A as the upgrade path.
- The body-text dependency tracking already works — we're only sacrificing label-fast-filter, not the semantic.
- It's the smallest change.

But I genuinely don't know how heavily `pulse-issue-reconcile-actions.sh` and the rest of the label-fast-path consumers rely on the `blocked-by:t<NNN>` label being present, vs. how often they fall back to body-text parsing. That's the question the maintainer can answer without me.

## Environment

- **aidevops version**: 3.13.5
- **AI Assistant**: OpenCode
- **OS**: Ubuntu 24.04.4 LTS
- **Shell**: bash 5.2.21(1)-release
- **Working repo**: robstiles/qs-agency
- **gh CLI**: gh version 2.91.0

## Additional Context

- This was hit during interactive Phase 4 review of an unrelated PR while filing a follow-up task with proper t1900 / t2417 worker-ready description (which included the `blocked-by:` reference that triggered auto-emission).
- See-also: sibling issue (TBD) — `_ensure_todo_entry_written` mega-line bug filed in the same session. Different bug, unrelated to this one — but both surfaced in the same `claim-task-id.sh` invocation.

<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.13.5 plugin for [OpenCode](https://opencode.ai) v1.14.28 with claude-opus-4-7 spent 40m and 105,507 tokens on this with the user in an interactive session.

--- a/.agents/scripts/claim-task-id.sh
+++ b/.agents/scripts/claim-task-id.sh
@@ -674,9 +674,37 @@ _extract_github_slug() {
 	return 0
 }
 
+# _auto_create_blocked_by_label — create a missing blocked-by:tNNN or blocked-by:#NNN
+# label in the repo when it does not yet exist. Updates the session label cache on success.
+# Args: $1 repo_slug (owner/repo), $2 label name (e.g. "blocked-by:t324"), $3 cache_file path.
+# Returns: 0 = label created (or already exists), 1 = creation failed (rate limit / permission).
+_auto_create_blocked_by_label() {
+	local repo_slug="$1"
+	local label="$2"
+	local cache_file="$3"
+	local predecessor="${label#blocked-by:}"
+
+	if gh label create "$label" --repo "$repo_slug" \
+		--color "5319E7" \
+		--description "Blocked by predecessor task ${predecessor}" \
+		>/dev/null 2>&1; then
+		# Append to session cache so subsequent lookups in this invocation hit
+		printf '%s\n' "$label" >>"$cache_file" 2>/dev/null || true
+		log_info "Auto-created label '${label}' in ${repo_slug} (t2823/t2800 auto-create exception)"
+		return 0
+	else
+		log_warn "Could not auto-create label '${label}' in ${repo_slug} (rate limit or permission) — proceeding without it"
+		return 1
+	fi
+}
+
 # _validate_labels_exist — check that every label in $2 exists in repo $1.
 # Args: $1 repo_slug (owner/repo), $2 comma-separated label names.
 # Returns: 0 = all valid (or fail-open), 1 = invalid labels found.
+#
+# Auto-create exception (GH#21474): labels matching ^blocked-by:(t[0-9]+|#[0-9]+)$
+# are auto-created when missing — they are a deterministic function of the predecessor
+# task ID and safe to create on demand. All other invalid labels still abort (exit 3).
 _validate_labels_exist() {
 	local repo_slug="$1"
 	local labels_csv="$2"
@@ -706,6 +734,10 @@ _validate_labels_exist() {
 		fi
 	fi
 
+	# Regex for the auto-create exception class (blocked-by:tNNN or blocked-by:#NNN).
+	# These labels are auto-created when missing — all other invalid labels still abort.
+	local _BLOCKED_BY_AUTO_CREATE_REGEX='^blocked-by:(t[0-9]+|#[0-9]+)$'
+
 	# Check each label against the cache
 	local invalid_labels=""
 	local label
@@ -721,6 +753,15 @@ _validate_labels_exist() {
 		label="${label%"${label##*[![:space:]]}"}"  # trim trailing whitespace
 		[[ -z "$label" ]] && continue
 		if ! grep -Fxq "$label" "$cache_file" 2>/dev/null; then
+			# Auto-create exception: blocked-by:tNNN / blocked-by:#NNN labels are
+			# synthesised by t2823 from description text and may not exist yet in fresh
+			# consumer repos. Create them on demand and continue — best-effort (t2800/GH#21474).
+			if [[ "$label" =~ $_BLOCKED_BY_AUTO_CREATE_REGEX ]]; then
+				_auto_create_blocked_by_label "$repo_slug" "$label" "$cache_file" || true
+				# Whether creation succeeded or not, don't abort — body text still records
+				# the dependency for pulse-dep-graph.sh and pulse-issue-reconcile-actions.sh.
+				continue
+			fi
 			if [[ -n "$invalid_labels" ]]; then
 				invalid_labels="${invalid_labels}, '${label}'"
 			else
@@ -730,8 +771,15 @@ _validate_labels_exist() {
 	done
 
 	if [[ -n "$invalid_labels" ]]; then
-		log_error "Pre-flight label validation failed — invalid label(s): ${invalid_labels}"
-		log_error "  Run: gh label list --repo \"${repo_slug}\" | grep -i '<name>'"
+		log_error "Pre-flight label validation failed — label(s) do not exist in this repo: ${invalid_labels}"
+		log_error ""
+		log_error "Each invalid label must be created before claiming, OR use one of these options:"
+		log_error "  (a) Create the label and re-run:"
+		log_error "        gh label create '<name>' --repo \"${repo_slug}\" --color '0075ca' --description '...'"
+		log_error "  (b) If the label was auto-emitted from a --description predecessor reference,"
+		log_error "      skip it: re-run with --no-blocked-by (dependency still recorded in body text)"
+		log_error "  (c) Use a different blocking label that already exists, e.g. 'status:blocked'"
+		log_error ""
 		log_error "  Claim aborted — counter NOT advanced."
 		return 1
 	fi

--- a/.agents/scripts/tests/test-claim-label-auto-create.sh
+++ b/.agents/scripts/tests/test-claim-label-auto-create.sh
@@ -1,0 +1,370 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: MIT
+# SPDX-FileCopyrightText: 2025-2026 Marcus Quinn
+#
+# test-claim-label-auto-create.sh — GH#21474 regression guard.
+#
+# Validates the auto-create exception in _validate_labels_exist(): when a label
+# matching ^blocked-by:(t[0-9]+|#[0-9]+)$ is missing from the repo, the helper
+# auto-creates it via gh label create and continues instead of aborting.
+#
+# Cases covered:
+#   1. blocked-by:t999 missing, gh label create succeeds   → label created in cache,
+#                                                             _validate_labels_exist returns 0
+#   2. blocked-by:t999 missing, gh label create fails       → warning only, returns 0
+#                                                             (best-effort / fail-open)
+#   3. tier:standrad (typo) missing                         → still aborts with return 1
+#                                                             (t2800 typo-protection preserved)
+#   4. blocked-by:#999 (issue-number variant) missing,
+#      gh label create succeeds                             → label created, returns 0
+#   5. blocked-by:t999 already exists in cache              → no create called, returns 0
+#
+# NOTE: not using `set -e` — assertions rely on capturing non-zero exits.
+
+set -uo pipefail
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)" || exit
+CLAIM_SCRIPT="${SCRIPT_DIR}/../claim-task-id.sh"
+
+RED=$'\033[0;31m'
+GREEN=$'\033[0;32m'
+NC=$'\033[0m'
+
+PASS=0
+FAIL=0
+ERRORS=""
+
+# ---------------------------------------------------------------------------
+# Test framework helpers
+# ---------------------------------------------------------------------------
+
+pass() {
+	local name="${1:-}"
+	printf '%s[PASS]%s %s\n' "$GREEN" "$NC" "$name"
+	PASS=$((PASS + 1))
+	return 0
+}
+
+fail() {
+	local name="${1:-}"
+	local detail="${2:-}"
+	printf '%s[FAIL]%s %s\n' "$RED" "$NC" "$name"
+	[[ -n "$detail" ]] && printf '       detail: %s\n' "$detail"
+	FAIL=$((FAIL + 1))
+	ERRORS="${ERRORS}\n  - ${name}: ${detail}"
+	return 0
+}
+
+assert_eq() {
+	local name="$1" got="$2" want="$3"
+	if [[ "$got" == "$want" ]]; then
+		pass "$name"
+	else
+		fail "$name" "want='${want}' got='${got}'"
+	fi
+	return 0
+}
+
+assert_return() {
+	local name="$1" actual_rc="$2" want_rc="$3"
+	if [[ "$actual_rc" == "$want_rc" ]]; then
+		pass "$name"
+	else
+		fail "$name" "want exit=${want_rc} got exit=${actual_rc}"
+	fi
+	return 0
+}
+
+# ---------------------------------------------------------------------------
+# Shared setup: source claim-task-id.sh with minimal stubs.
+# The BASH_SOURCE guard in the script prevents main() from running on source.
+# ---------------------------------------------------------------------------
+
+# Create a temp dir for stubs and state shared across all tests
+BASE_STUB_DIR=""
+BASE_STUB_DIR=$(mktemp -d)
+trap 'rm -rf '"$BASE_STUB_DIR"'' EXIT
+
+# Fake git: no-op (silences detect_platform remote calls)
+cat >"${BASE_STUB_DIR}/git" <<'STUB'
+#!/usr/bin/env bash
+exit 0
+STUB
+chmod +x "${BASE_STUB_DIR}/git"
+
+# Fake jq: no-op
+cat >"${BASE_STUB_DIR}/jq" <<'STUB'
+#!/usr/bin/env bash
+exit 0
+STUB
+chmod +x "${BASE_STUB_DIR}/jq"
+
+export HOME="${BASE_STUB_DIR}/home"
+mkdir -p "${HOME}/.aidevops/logs"
+
+# ---------------------------------------------------------------------------
+# Helper: build a per-test gh stub in a new subdir and prepend it to PATH.
+# The stub file is written by the caller to configure per-test behavior.
+# ---------------------------------------------------------------------------
+_make_stub_dir() {
+	local stub_dir
+	stub_dir=$(mktemp -d "${BASE_STUB_DIR}/test-XXXXXX")
+	# Copy shared stubs
+	cp "${BASE_STUB_DIR}/git" "${stub_dir}/git"
+	cp "${BASE_STUB_DIR}/jq" "${stub_dir}/jq"
+	printf '%s' "$stub_dir"
+	return 0
+}
+
+# ---------------------------------------------------------------------------
+# Source the script once (functions are loaded into shell; main() is guarded)
+# ---------------------------------------------------------------------------
+# We need a gh stub available at source time for the auth check:
+cat >"${BASE_STUB_DIR}/gh" <<'STUB'
+#!/usr/bin/env bash
+if [[ "${1:-}" == "auth" && "${2:-}" == "status" ]]; then exit 0; fi
+exit 0
+STUB
+chmod +x "${BASE_STUB_DIR}/gh"
+export PATH="${BASE_STUB_DIR}:${PATH}"
+
+# shellcheck disable=SC1090
+if ! source "$CLAIM_SCRIPT" 2>/dev/null; then
+	printf '%s[FATAL]%s Failed to source %s\n' "$RED" "$NC" "$CLAIM_SCRIPT" >&2
+	exit 1
+fi
+
+# ---------------------------------------------------------------------------
+# Test 1: blocked-by:t999 missing, gh create succeeds → returns 0, label in cache
+# ---------------------------------------------------------------------------
+_run_test1() {
+	local stub_dir
+	stub_dir=$(_make_stub_dir)
+
+	# Track whether gh label create was invoked
+	local create_called_file="${stub_dir}/create_called"
+
+	cat >"${stub_dir}/gh" <<STUB
+#!/usr/bin/env bash
+if [[ "\${1:-}" == "auth" && "\${2:-}" == "status" ]]; then exit 0; fi
+if [[ "\${1:-}" == "label" && "\${2:-}" == "list" ]]; then
+  # Return known labels (not including blocked-by:t999)
+  printf 'tier:standard\nauto-dispatch\nbug\n'
+  exit 0
+fi
+if [[ "\${1:-}" == "label" && "\${2:-}" == "create" ]]; then
+  printf '%s' "called" > "${create_called_file}"
+  exit 0
+fi
+exit 0
+STUB
+	chmod +x "${stub_dir}/gh"
+
+	# Temporarily prepend our per-test stub
+	local saved_path="$PATH"
+	PATH="${stub_dir}:${PATH}"
+
+	# Reset session cache so the function fetches fresh
+	local saved_cache="${AIDEVOPS_LABEL_CACHE_FILE:-}"
+	unset AIDEVOPS_LABEL_CACHE_FILE
+
+	local rc=0
+	_validate_labels_exist "owner/repo" "tier:standard,blocked-by:t999" 2>/dev/null || rc=$?
+
+	PATH="$saved_path"
+	[[ -n "$saved_cache" ]] && export AIDEVOPS_LABEL_CACHE_FILE="$saved_cache" || true
+
+	assert_return "test1_blocked_by_create_succeeds_returns_0" "$rc" "0"
+
+	local create_called=""
+	[[ -f "$create_called_file" ]] && create_called="$(cat "$create_called_file")"
+	assert_eq "test1_gh_label_create_was_called" "$create_called" "called"
+
+	return 0
+}
+
+# ---------------------------------------------------------------------------
+# Test 2: blocked-by:t999 missing, gh create fails → still returns 0 (best-effort)
+# ---------------------------------------------------------------------------
+_run_test2() {
+	local stub_dir
+	stub_dir=$(_make_stub_dir)
+
+	cat >"${stub_dir}/gh" <<'STUB'
+#!/usr/bin/env bash
+if [[ "${1:-}" == "auth" && "${2:-}" == "status" ]]; then exit 0; fi
+if [[ "${1:-}" == "label" && "${2:-}" == "list" ]]; then
+  printf 'tier:standard\nauto-dispatch\nbug\n'
+  exit 0
+fi
+if [[ "${1:-}" == "label" && "${2:-}" == "create" ]]; then
+  # Simulate permission/rate-limit failure
+  exit 1
+fi
+exit 0
+STUB
+	chmod +x "${stub_dir}/gh"
+
+	local saved_path="$PATH"
+	PATH="${stub_dir}:${PATH}"
+
+	local saved_cache="${AIDEVOPS_LABEL_CACHE_FILE:-}"
+	unset AIDEVOPS_LABEL_CACHE_FILE
+
+	local rc=0
+	_validate_labels_exist "owner/repo" "tier:standard,blocked-by:t999" 2>/dev/null || rc=$?
+
+	PATH="$saved_path"
+	[[ -n "$saved_cache" ]] && export AIDEVOPS_LABEL_CACHE_FILE="$saved_cache" || true
+
+	assert_return "test2_blocked_by_create_fails_still_returns_0" "$rc" "0"
+
+	return 0
+}
+
+# ---------------------------------------------------------------------------
+# Test 3: tier:standrad (typo) still aborts — typo-protection preserved (t2800)
+# ---------------------------------------------------------------------------
+_run_test3() {
+	local stub_dir
+	stub_dir=$(_make_stub_dir)
+
+	cat >"${stub_dir}/gh" <<'STUB'
+#!/usr/bin/env bash
+if [[ "${1:-}" == "auth" && "${2:-}" == "status" ]]; then exit 0; fi
+if [[ "${1:-}" == "label" && "${2:-}" == "list" ]]; then
+  printf 'tier:standard\nauto-dispatch\nbug\n'
+  exit 0
+fi
+exit 0
+STUB
+	chmod +x "${stub_dir}/gh"
+
+	local saved_path="$PATH"
+	PATH="${stub_dir}:${PATH}"
+
+	local saved_cache="${AIDEVOPS_LABEL_CACHE_FILE:-}"
+	unset AIDEVOPS_LABEL_CACHE_FILE
+
+	local rc=0
+	_validate_labels_exist "owner/repo" "tier:standrad,bug" 2>/dev/null || rc=$?
+
+	PATH="$saved_path"
+	[[ -n "$saved_cache" ]] && export AIDEVOPS_LABEL_CACHE_FILE="$saved_cache" || true
+
+	assert_return "test3_typo_label_still_aborts" "$rc" "1"
+
+	return 0
+}
+
+# ---------------------------------------------------------------------------
+# Test 4: blocked-by:#999 (issue-number variant), gh create succeeds → returns 0
+# ---------------------------------------------------------------------------
+_run_test4() {
+	local stub_dir
+	stub_dir=$(_make_stub_dir)
+	local create_called_file="${stub_dir}/create_called"
+
+	cat >"${stub_dir}/gh" <<STUB
+#!/usr/bin/env bash
+if [[ "\${1:-}" == "auth" && "\${2:-}" == "status" ]]; then exit 0; fi
+if [[ "\${1:-}" == "label" && "\${2:-}" == "list" ]]; then
+  printf 'tier:standard\nauto-dispatch\nbug\n'
+  exit 0
+fi
+if [[ "\${1:-}" == "label" && "\${2:-}" == "create" ]]; then
+  printf '%s' "called" > "${create_called_file}"
+  exit 0
+fi
+exit 0
+STUB
+	chmod +x "${stub_dir}/gh"
+
+	local saved_path="$PATH"
+	PATH="${stub_dir}:${PATH}"
+
+	local saved_cache="${AIDEVOPS_LABEL_CACHE_FILE:-}"
+	unset AIDEVOPS_LABEL_CACHE_FILE
+
+	local rc=0
+	_validate_labels_exist "owner/repo" "bug,blocked-by:#999" 2>/dev/null || rc=$?
+
+	PATH="$saved_path"
+	[[ -n "$saved_cache" ]] && export AIDEVOPS_LABEL_CACHE_FILE="$saved_cache" || true
+
+	assert_return "test4_blocked_by_hash_variant_returns_0" "$rc" "0"
+
+	local create_called=""
+	[[ -f "$create_called_file" ]] && create_called="$(cat "$create_called_file")"
+	assert_eq "test4_gh_label_create_was_called_for_hash_variant" "$create_called" "called"
+
+	return 0
+}
+
+# ---------------------------------------------------------------------------
+# Test 5: blocked-by:t999 already in cache → gh label create NOT called, returns 0
+# ---------------------------------------------------------------------------
+_run_test5() {
+	local stub_dir
+	stub_dir=$(_make_stub_dir)
+	local create_called_file="${stub_dir}/create_called"
+
+	cat >"${stub_dir}/gh" <<STUB
+#!/usr/bin/env bash
+if [[ "\${1:-}" == "auth" && "\${2:-}" == "status" ]]; then exit 0; fi
+if [[ "\${1:-}" == "label" && "\${2:-}" == "list" ]]; then
+  # Include blocked-by:t999 so it's already in the cache
+  printf 'tier:standard\nauto-dispatch\nbug\nblocked-by:t999\n'
+  exit 0
+fi
+if [[ "\${1:-}" == "label" && "\${2:-}" == "create" ]]; then
+  printf '%s' "called" > "${create_called_file}"
+  exit 0
+fi
+exit 0
+STUB
+	chmod +x "${stub_dir}/gh"
+
+	local saved_path="$PATH"
+	PATH="${stub_dir}:${PATH}"
+
+	local saved_cache="${AIDEVOPS_LABEL_CACHE_FILE:-}"
+	unset AIDEVOPS_LABEL_CACHE_FILE
+
+	local rc=0
+	_validate_labels_exist "owner/repo" "bug,blocked-by:t999" 2>/dev/null || rc=$?
+
+	PATH="$saved_path"
+	[[ -n "$saved_cache" ]] && export AIDEVOPS_LABEL_CACHE_FILE="$saved_cache" || true
+
+	assert_return "test5_blocked_by_already_in_cache_returns_0" "$rc" "0"
+
+	local create_called=""
+	[[ -f "$create_called_file" ]] && create_called="$(cat "$create_called_file")"
+	assert_eq "test5_no_create_call_when_label_exists" "$create_called" ""
+
+	return 0
+}
+
+# ---------------------------------------------------------------------------
+# Run all tests
+# ---------------------------------------------------------------------------
+printf '\n=== test-claim-label-auto-create.sh (GH#21474) ===\n\n'
+
+_run_test1
+_run_test2
+_run_test3
+_run_test4
+_run_test5
+
+# ---------------------------------------------------------------------------
+# Summary
+# ---------------------------------------------------------------------------
+printf '\n%d tests run: %d passed, %d failed\n' "$((PASS + FAIL))" "$PASS" "$FAIL"
+
+if [[ "$FAIL" -gt 0 ]]; then
+	printf '\nFailed tests:%b\n' "$ERRORS"
+	exit 1
+fi
+
+exit 0

